internal/statemachine/state_machine.go

```diff
@@ -0,0 +1,323 @@
+package statemachine
+
+import (
+	"encoding/gob"
+	"fmt"
+	"os"
+	"strconv"
+)
+
+// used for --resume to convert a named step to an ordinal
+var stateNames = map[string]int{
```

Comment:
These tables make me nervous that someone will come along and try to insert a new state in between 6 and 7 and not consistently number them or duplicate the numbers or something similarly hard to catch by visually comparing the tables.

What's your thoughts on having the initial data be a mapping from the string name to the function 
(which should read well from a code review standpoint - 
`"make_temporary_directories": (*StateMachine).makeTemporaryDirectories`)
and then iterating over this table to generate the stateNames and stateFuncs maps?  This has a side benefit that if you have to add a new state in the middle it becomes a one-line change and you won't get the renumbering wrong.

Reply:
Per a conversation with sil2100, we're going to change up how these tables work altogether in a way that should address this as a side-effect.